auth-service/src/test/java/com/petclinic/auth/e2e/User/AuthServiceE2ETests.java

```diff
@@ -190,10 +193,59 @@ void admin_access_protected_route() throws Exception {
                 .andExpect(jsonPath("$.totalPages").value(ceil(userRepo.count() / 10.0)));
     }
 
-    private void registerUser() throws Exception {
+    @Test
+    @DisplayName("Given duplicate email, expect bad request exception with sensical message")
+    void duplicate_email_register() throws Exception {
+
+        registerUser();
 
         final String asString = objectMapper.writeValueAsString(ID_LESS_USER);
         mockMvc.perform(post("/users").contentType(APPLICATION_JSON).content(asString))
+                .andExpect(status().is4xxClientError())
+                .andExpect(content().contentType(APPLICATION_JSON))
+                .andExpect(jsonPath("$.message").exists())
+                .andExpect(jsonPath("$.timestamp").exists())
+                .andExpect(jsonPath("$.statusCode").exists());
+    }
+
+    @Test
+    @DisplayName("Given non-registered user login, return 401")
+    void non_registered_user() throws Exception {
+
+        final String asString = objectMapper.writeValueAsString(new HashMap<String, String>(){{
+            put("email", USER.getEmail());
+            put("password", USER.getPassword());
+        }});
+
+        final MvcResult result = mockMvc.perform(post("/users/login").contentType(APPLICATION_JSON).content(asString))
+                .andExpect(status().is4xxClientError())
+                .andExpect(content().contentType(APPLICATION_JSON))
+                .andExpect(jsonPath("$.password").doesNotExist())
+                .andExpect(jsonPath("$.statusCode").value(UNAUTHORIZED.value()))
+                .andExpect(jsonPath("$.timestamp").exists())
+                .andExpect(jsonPath("$.message").value(format("Password not valid for email %s", USER.getEmail())))
```

Comment:
Shouldn't this expect a message saying that there is no user registered with the given email?